R/pooling.R

```diff
@@ -70,22 +70,35 @@ effective_samplesize <- function(
 #' \code{last_date} - period will be considered for each pollster. Defaults
 #' to current date.
 #' @param period See \code{last_date} argument.
+#' @param period_extended Optional. If specified, all surveys in the time-window
+#' from \code{last_date} - period_extended to \code{last_date} - period will
+#' also be considered for each pollster, but only after downweighting them by
+#' halving their true sample size.
 #' @import dplyr checkmate
 #' @keywords internal
 get_eligible <- function(
   surveys,
   pollsters,
-  last_date = Sys.Date(),
-  period    = 14) {
+  last_date       = Sys.Date(),
+  period          = 14,
+  period_extended = NA) {
 
   assert_data_frame(surveys, min.rows = 2, min.cols = 2)
   assert_date(last_date)
   assert_character(pollsters, null.ok = TRUE)
   assert_number(period, lower = 1, finite = TRUE)
-
+  assert_number(period_extended, lower = 1, finite = TRUE, na.ok = TRUE)
+  assert_true(period < period_extended, na.ok = TRUE)
+  
+  first_date <- last_date - ifelse(!is.na(period_extended), period_extended, period)
+  
   surveys %>% filter(.data$pollster %in% pollsters) %>%
     unnest(surveys) %>%
-    filter(date >= last_date - period & date <= last_date) %>%
+    filter(date >= first_date & date <= last_date) %>%
+    unnest() %>%
+    mutate(respondents = respondents * ifelse(date >= last_date - period, 1, 0.5),
+           votes = votes * ifelse(date >= last_date - period, 1, 0.5)) %>%
+    nest(-pollster, -date, -start, -end, -respondents) %>%
```

Comment:
things like that will show up as "no visible binding ..." in devtools::check().
Avoid using one_of as here: https://github.com/adibender/coalitions/blob/master/R/survey-helpers.R#L50
Can also be used as one_of(...)